students/rhamersky/Lesson_10/mailroom_part6.py

```diff
@@ -0,0 +1,392 @@
+
+        # Creates the header for the chart
+        for heading in lst_headers:
+            print(("{0:{padding}}".format(heading, padding=longest_entry)), end=" ")
+        print()
+
+        # Creates the donor rows for the chart.
+        sorted_dic = Data.dic_sort(dic_donors)
+        for name in sorted_dic:
+            donation_avg = dic_donors[name][0] / dic_donors[name][1]
+            print(("{0:{padding}}".format(name, padding=longest_entry)), end=" ")
+            print(("{0:{padding}}".format("{}".format(dic_donors[name][1]), padding=longest_entry)), end=" ")
+            print(("{0:{padding}}".format("${:,.2f}".format(donation_avg), padding=longest_entry)), end=" ")
+            print(("{0:{padding}}".format("${:,.2f}".format(dic_donors[name][0]), padding=longest_entry)), end=" ")
+            print()
+
+    @staticmethod
+    def challenge(str_past_donations, name_of_donor, str_filter,  factor):
+        '''
+        This function will increase all the donations by a factor.
+        :param factor: The number to multiply the donations by.
+        :return: Returns the total factored amount.
+        '''
+        lst_values = []
+        new_factored_donation = int()
+        # Checks to see if there are past donations from the donor.
+        if str_past_donations == "No":
+            lst_values.append(dic_donors[name_of_donor][0])
+        else:
+            for index in range(2, len(dic_donors[name_of_donor])):
+                lst_values.append(dic_donors[name_of_donor][index])
+
+        # Filters donations
+        if str_filter == "Max":
+            lst_values = filter(lambda filter_value: filter_value >= 2500, lst_values)
+        else:
+            lst_values = filter(lambda filter_value: filter_value < 2500, lst_values)
+
+        # Adds a factor to the filtered donations.
+        factored_donations = map(lambda donation: donation * factor, lst_values)
+
+        # Sums factored donations
+        for past_donation in list(factored_donations):
+            new_factored_donation += past_donation
+
+        return print(name_of_donor + ", could have donated ${:,.2f}.".format(new_factored_donation))
+
+def main():
+    '''
+    This is the main function. This will run the full program.
+    :return:
+    '''
+
+    while True:
+        # User menu to update the priority list.
+        print()  # --> Formatting
+        print("\t\tMenu of Options\n\n"
+              "\t\t1) Send a Thank You\n\n"
+              "\t\t2) Create a Report\n\n"
+              "\t\t3) Send letters to everyone\n\n"
+              "\t\t4) Projection\n\n"
+              "\t\t5) Quit\n")
+
+        # Error handles user's option choice.
+        int_user_input = Menu.menu_options()
+
+        # User selects option 1. Can display a list of the donors and where the user can add donors and donations.
+        if int_user_input == 1:
+            print()  # --> Formatting
+            Menu.send_thankyou()
+            continue
```

Comment:
You should be using a dict to switch between user selections at this point per the instructions in Lesson 4. This would really simplify this section of code as well as some of the related validation logic for use input